### PR TITLE
lab5

### PR DESCRIPTION
**Pull request by Dima1234566**
-

### CODE REVIEW

File: lab5/app/build.gradle [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'com.android.support:appcompat-v7:28.0.0'
```
Comment (iRafal) [inline lab5/app/build.gradle:23]: fix warning

File: lab5/app/src/main/java/e/dimas/lab5/Api.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package e.dimas.lab5;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.http.GET;
+
+public interface Api {
+
+    String Base_URL = "http://www.amiiboapi.com/api/amiibo/";
```
Comment (iRafal) [inline lab5/app/src/main/java/e/dimas/lab5/Api.java]: static final

File: lab5/app/src/main/java/e/dimas/lab5/Api.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package e.dimas.lab5;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.http.GET;
+
+public interface Api {
```
Comment (iRafal) [inline lab5/app/src/main/java/e/dimas/lab5/Api.java:6]: too much empty lines.

File: lab5/.gitignore [hunk 4]
```diff
@@ -0,0 +1,11 @@
+*.iml
+.gradle
+/local.properties
+/.idea/caches/build_file_checksums.ser
+/.idea/libraries
+/.idea/modules.xml
+/.idea/workspace.xml
```
Comment (MarkoKuspis) [inline lab5/.gitignore]: /.idea/*

File: lab5/app/build.gradle [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+    testImplementation 'junit:junit:4.12'
+    implementation 'com.squareup.retrofit2:retrofit:2.4.0'
+    implementation 'com.squareup.retrofit2:converter-gson:2.0.0-beta4'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
```
Comment (MarkoKuspis) [inline lab5/app/build.gradle]: You don't use any of two libraries above

File: lab5/app/build.gradle [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+}
+
+dependencies {
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'com.android.support:appcompat-v7:28.0.0'
+    implementation 'com.android.support.constraint:constraint-layout:1.1.3'
+    testImplementation 'junit:junit:4.12'
```
Comment (MarkoKuspis) [inline lab5/app/build.gradle]: You don't use it as well

File: lab5/app/src/androidTest/java/e/dimas/lab5/ExampleInstrumentedTest.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+@RunWith(AndroidJUnit4.class)
+public class ExampleInstrumentedTest {
+    @Test
+    public void useAppContext() {
+        // Context of the app under test.
+        Context appContext = InstrumentationRegistry.getTargetContext();
+
+        assertEquals("e.dimas.lab5", appContext.getPackageName());
+    }
+}
```
Comment (MarkoKuspis) [inline lab5/app/src/androidTest/java/e/dimas/lab5/ExampleInstrumentedTest.java]: Drop this file

File: lab5/app/src/main/java/e/dimas/lab5/MainActivity.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+    public class MainActivity extends AppCompatActivity {
+        ListView listView;
+        @Override
```
Comment (MarkoKuspis) [inline lab5/app/src/main/java/e/dimas/lab5/MainActivity.java:20]: Indentation for this file is incorrect

File: lab5/app/src/main/java/e/dimas/lab5/MainActivity.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+    public class MainActivity extends AppCompatActivity {
+        ListView listView;
```
Comment (MarkoKuspis) [inline lab5/app/src/main/java/e/dimas/lab5/MainActivity.java]: private

File: lab5/app/src/main/java/e/dimas/lab5/Photo.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
+package e.dimas.lab5;
+
+
+public class Photo {
+
+    private String name;
```
Comment (MarkoKuspis) [inline lab5/app/src/main/java/e/dimas/lab5/Photo.java]:
```suggestion
    private final String name;
```

**Review by MarkoKuspis — CHANGES_REQUESTED** [review]

**Review by MarkoKuspis — APPROVED** [review]

**Pull request merged**